File: Java - old/app/src/main/java/com/stutzel/bateria/Monitor.java
package com.stutzel.bateria;

import android.app.NotificationManager;
import android.app.PendingIntent;
import android.app.Service;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.BatteryManager;
import android.os.IBinder;
import android.support.v4.app.NotificationCompat;

public class Monitor extends Service {

    private static final int Id = 16111994;
    private NotificationCompat.Builder mBuilder;
    private NotificationManager mNotificationManager;
    private final BroadcastReceiver mBatInfoReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context arg0, Intent intent) {
            int level = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
            int scale = intent.getIntExtra(BatteryManager.EXTRA_SCALE, -1);
            int batteryPct = (int) ((level / (float) scale) * 100);
            notificacao(batteryPct);
        }
    };


    public Monitor() {
    }

    private void notificacao(int level) {
        mBuilder.setSmallIcon(getResources().getIdentifier("b" + level, "drawable", "com.stutzel.bateria"))
                .setContentText("Seu nível de bateria é: " + level + "%");
        mNotificationManager.notify(Id, mBuilder.build());
    }

    private void criaNotificacao() {
        mBuilder = new NotificationCompat.Builder(this)
                .setSmallIcon(R.drawable.ic_launcher)
                .setContentTitle("Bateria")
                .setContentText("Seu nível de bateria é:")
                .setOngoing(true)
                .setPriority(-1);

        PendingIntent contentIntent = PendingIntent.getActivity(this, 0,
                new Intent(this, Bateria.class), 0);

        mBuilder.setAutoCancel(false);
        mBuilder.setContentIntent(contentIntent);
        mNotificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        mNotificationManager.notify(Id, mBuilder.build());
    }

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    public void onCreate() {
        super.onCreate();
        criaNotificacao();
        registerReceiver(this.mBatInfoReceiver, new IntentFilter(Intent.ACTION_BATTERY_CHANGED));
    }

    public void onDestroy() {
        unregisterReceiver(this.mBatInfoReceiver);
        mNotificationManager.cancel(Id);
    }
}

File: Kotlin-new/app/src/main/java/br/stutzel/bateria/Monitor.kt
package br.stutzel.bateria

import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.app.Service
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.graphics.Color
import android.os.Build
import android.os.IBinder
import android.support.v4.app.NotificationCompat


class Monitor : Service() {
    private var Id = 16111994
    private lateinit var mBuilder: NotificationCompat.Builder
    private lateinit var mNotificationManager: NotificationManager
    private var mBatInfoReceiver = BatteryReceiver(this)


    override fun onBind(intent: Intent): IBinder? = null
    override fun onCreate() {
        super.onCreate()
        buildNotification()
        registerReceiver(mBatInfoReceiver, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
    }

    override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(mBatInfoReceiver)
        cleanNotification()
    }

    private fun cleanNotification() {
        mNotificationManager.cancel(Id)
    }

    private fun buildNotification() {
        mNotificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        val channelId = "default"
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            mNotificationManager.createNotificationChannel(NotificationChannel(channelId, "Bateria", NotificationManager.IMPORTANCE_LOW))
        }
        mBuilder = NotificationCompat.Builder(this, channelId)
        mBuilder.setSmallIcon(R.mipmap.ic_launcher)
        mBuilder.setContentTitle(getString(R.string.notification_title))
        mBuilder.setContentText(getString(R.string.notification_text, 1))
        mBuilder.setOngoing(true)
        mBuilder.priority = -1
        mBuilder.setAutoCancel(false)
        mBuilder.setContentIntent(PendingIntent.getActivity(this, 0,
                Intent(this, MainActivity::class.java), 0))

        startForeground(Id, mBuilder.build())
    }

    fun updateNotification(batteryPct: Int) {
        mBuilder
                .setSmallIcon(generateSmallIcon(batteryPct))
                .setColor(Color.BLACK)
                .setContentText(getString(R.string.notification_text, batteryPct))
        mNotificationManager.notify(Id, mBuilder.build())
    }

    private fun generateSmallIcon(batteryPct: Int): Int {
        return resources.getIdentifier("b" + batteryPct, "drawable", "br.stutzel.bateria")
    }
}
